docs/sources/reference/components/loki/loki.source.file.md

```diff
@@ -105,6 +109,27 @@ If no file changes are detected, the poll frequency doubles until a file change
 
 If file changes are detected, the poll frequency is reset to `min_poll_frequency`.
 
+### `file_match`
+
+The `file_match` block enables built-in file discovery directly within `loki.source.file`.
+When enabled, the component discovers files on the local filesystem using glob patterns and the [doublestar][] library.
+
+The following arguments are supported:
+
+| Name                | Type                | Description                                                 | Default | Required |
+|---------------------|---------------------|-------------------------------------------------------------|---------|----------|
+| `enabled`           | `bool`              | Enable file discovery using glob patterns.                  | `false` | no       |
+| `ignore_older_than` | `duration`          | Ignores files with modification times before this duration. | `"0s"`  | no       |
+| `sync_period`       | `duration`          | How often to sync filesystem and targets.                   | `"10s"` | no       |
+
+Benefits of using `file_match` over `local.file_match`:
+- Simplified configuration: No need for a separate `local.file_match` component
+- Better performance: File discovery is integrated directly into `loki.source.file`, eliminating the overhead of component-to-component communication
+- Same functionality: Supports the same glob patterns and features, including `__path_exclude__` for excluding files
+
+When `enabled` is set to `true`, you can use glob patterns (e.g., `/tmp/*.log` or `/var/log/**/*.log`) directly in the `targets` argument's `__path__` label.
```

Comment:
What happens when enabled is false?

Reply:
Can we write that in docs?

Reply:
We already have it here https://github.com/grafana/alloy/pull/4743/files#diff-4b8c6b16f7e932bea249731a9f49d947024699eb129a37f0846aed9c1086e945R21-R23.

Do you think we need it in both places?